tests/Language/Java/InlineSpec.hs

```diff
@@ -52,6 +56,11 @@ spec = do
           _ :: JJObject <- [java| $obj1 |]
           return ()
 
+        it "Supports parameterized type synonyms" $ do
+          obj :: List ObjectClass <- [java| new Object() {} |]
```

Comment:
I think that's a good idea, yes. We could conceivably provide a "checked" and an "unchecked" mode.